### PR TITLE
Add public missing asyncio stubs for windows and proactor files

### PR DESCRIPTION
**Pull request by CraftSpider**
More work towards #2371.

Adds in all the `windows_*.py` and `proactor_*.py` files. There are mypy test failures, but I don't want to add stubs for those methods without discussion because the real implementation doesn't actually have them. `create_unix_*` are not actual methods on windows-specific asyncio loops.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Sorry for the delay in reviewing this PR! The CI failures obviously need to be dealt with. The best option is probably to add these methods to the stub with a comment that they don't really exist at runtime.

**Comment by CraftSpider** [comment]
Alright, after a merge and fixes, build is now passing. Added the comment about runtime status of the methods in BaseProactorEventLoop

File: stdlib/3/asyncio/windows_utils.pyi [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+def pipe(*, duplex: bool = ..., overlapped: Tuple[bool, bool] = ..., bufsize: int = ...) -> Tuple[int, int]: ...
+
+class PipeHandle:
+
+    def __init__(self, handle: int) -> None: ...
+    def __repr__(self) -> str: ...
+    def __del__(self) -> None: ...
+    def __enter__(self) -> PipeHandle: ...
+    def __exit__(self, t: type, v: BaseException, tb: TracebackType) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/asyncio/windows_utils.pyi]: Strictly all the arguments should be Optional.

File: stdlib/3/asyncio/windows_events.pyi [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    def __del__(self) -> None: ...
+    def closed(self) -> bool: ...
+    def close(self) -> None: ...
+
+class _WindowsSelectorEventLoop(selector_events.BaseSelectorEventLoop): ...
+
+class ProactorEventLoop(proactor_events.BaseProactorEventLoop):
+
+    def __init__(self, proactor: IocpProactor = ...) -> None: ...
+    async def create_pipe_connection(self, protocol_factory: Callable[[], events.AbstractEventLoopPolicy], address: str) -> Tuple[proactor_events._ProactorDuplexPipeTransport, events.AbstractEventLoopPolicy]: ...
```
[the remark this answers is not in the document]
Comment (CraftSpider) [reply stdlib/3/asyncio/windows_events.pyi]: Pretty sure you're right, it actually expects a StreamReaderProtocol from what I can tell. Not sure why I put it as `AbstractEventLoopPolicy`

**Pull request merged**